src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/service/LaoService.kt

```diff
@@ -0,0 +1,31 @@
+package uk.gov.justice.digital.hmpps.hmppsallocations.service
+
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsallocations.client.WorkforceAllocationsToDeliusApiClient
+import uk.gov.justice.digital.hmpps.hmppsallocations.client.dto.DeliusCrnRestrictions
+
+@Service
+class LaoService(
+  @Qualifier("workforceAllocationsToDeliusApiClientUserEnhanced")
+  private val workforceAllocationsToDeliusApiClient: WorkforceAllocationsToDeliusApiClient,
+) {
+
+  suspend fun getCrnRestrictions(crn: String): DeliusCrnRestrictions {
+    val apopUsers = workforceAllocationsToDeliusApiClient.getApopUsers()
+    val staffCodes = apopUsers.map { it.staffCode }
```

Comment:
apopUsersStaffCodes

Reply:
Have done the suggested refactors and a couple of others.